xcodeproj/internal/resources.bzl

```diff
@@ -343,6 +343,19 @@ def collect_resources(
         parent.dependency_paths.append(child_bundle_path)
 
     frozen_bundles = []
+    resource_to_owners = {}
+    for (resource, owner) in resource_info.owners.to_list():
+        # AppleResourceInfo treats resource files inside xcassets separately, whereas
+        # rules_xcodeproj only cares about .xcassets folder
+        if ".xcassets" in resource:
+            resource = resource.split(".xcassets")[0] + ".xcassets"
+
+        # a resource can have multiple owners
+        if resource in resource_to_owners and owner not in resource_to_owners[resource]:
```

Comment:
Instead of `owner not in resource_to_owners[resource]`, you could make the value of the `dict` a set (a `dict` with keys as the values storing, and `None` as the value`), and then just insert without condition. Also, instead of `if resource in resource_to_owners` you can use `setdefault()`.

Reply:
good point!